examples/rquickjs-cli/src/main.rs

```diff
@@ -0,0 +1,42 @@
+use std::io::{stdout, Write};
+
+use rquickjs::{Context, Function, Object, Result, Runtime, Value};
+
+fn print(s: String) {
+    println!("{s}");
+}
+
+fn main() -> Result<()> {
+    let rt = Runtime::new()?;
+    let ctx = Context::full(&rt)?;
+
+    ctx.with(|ctx| -> Result<()> {
+        let global = ctx.globals();
+        global.set(
+            "print",
+            Function::new(ctx.clone(), print)?.with_name("print")?,
+        )?;
+        ctx.eval::<(), _>(
+            r#"
+globalThis.console = {
+  log(v){
+    globalThis.print(`${v}`);
+  }
+}
+"#,
+        )?;
+
+        let console: Object = global.get("console")?;
+        let js_log: Function = console.get("log")?;
+        loop {
+            let mut input = String::new();
+            let _ = stdout().write(b"> ")?;
+            stdout().flush()?;
+            std::io::stdin().read_line(&mut input)?;
+            let ret = ctx.eval::<Value, _>(input.as_bytes())?;
```

Comment:
This won't report errors properly. You should retrieve the error thrown by QuickJS by calling for example `.catch(&ctx)` on the returned result.

Reply:
Is this the right way to use it?
```rust
  match ctx.eval::<Value, _>(input.as_bytes()).catch(&ctx) {
      Ok(ret) => {
          js_log.call::<(Value<'_>,), ()>((ret,))?;
      }
      Err(err) => {
          println!("{err}");
      }
  }
```

![image](https://github.com/user-attachments/assets/541cddc1-5c4b-4cd6-a10d-b5adfb2c6cd9)

Reply:
Yes that will return the right error.